apps/new-dealer/controllers/supporting-documents-add-another.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+const BaseController = require('../../common/controllers/base');
+const path = require('path');
+
+module.exports = class UploadController extends BaseController {
+  get(req, res, callback) {
+    if (req.params.action === 'delete' && req.params.id) {
+      return this.removeItem(req, res);
+    }
+    return super.get(req, res, callback);
+  }
+
+  removeItem(req, res) {
+    const items = req.sessionModel.get('supporting-documents') || [];
+    req.sessionModel.set('supporting-documents', items.filter((file) => file.id !== req.params.id));
+    res.redirect(path.join(req.baseUrl, this.options.route));
```

Comment:
I don't think this is a suitable method for this line of code, because it has nothing to do with removing the item.